src/components/AuthForm.js

```diff
@@ -512,6 +513,51 @@ export default class AuthForm extends PureComponent<
     }
   };
 
+  componentDidUpdate(prevProps: AuthFormProps, prevState: AuthFormState) {
+    const {
+      formData: { server: prevServer },
+    } = prevState;
+    const {
+      formData: { server: newServer },
+    } = this.state;
+
+    if (prevServer !== newServer) {
+      // Server changed, set the authType back to "anonymous"
+      return this.onChange({
+        ...this.state,
+        formData: { ...this.state.formData, authType: ANONYMOUS_AUTH },
+      });
+    }
+
+    const {
+      session: {
+        serverInfo: { capabilities: prevCapabilities },
+      },
+    } = prevProps;
+    const {
+      session: {
+        serverInfo: { capabilities: newCapabilities },
+      },
+    } = this.props;
+    if (JSON.stringify(prevCapabilities) !== JSON.stringify(newCapabilities)) {
```

Comment:
Maybe here we should only compare the list of capabilities keys?

Reply:
Actually, I believe we should simply test that the previous capabilities where empty, which is the default value when changing a server (and reinitializing the capabilities)